pkg/server/plugin/nodeattestor/aws/session.go

```diff
@@ -33,13 +38,28 @@ func (cfg *SessionConfig) Validate(defaultAccessKeyID, defaultSecretAccessKey st
 }
 
 // newAWSSession create an AWS Session from the config and given region
-func newAWSSession(accessKeyID, secretAccessKey, region string) (*session.Session, error) {
+func newAWSSession(accessKeyID, secretAccessKey, region, asssumeRoleArn string) (*session.Session, error) {
 	var awsConf *aws.Config
 	if secretAccessKey != "" && accessKeyID != "" {
 		creds := credentials.NewStaticCredentials(accessKeyID, secretAccessKey, "")
 		awsConf = &aws.Config{Credentials: creds, Region: &region}
 	} else {
 		awsConf = &aws.Config{Region: &region}
 	}
+
+	// Optional: Assuming role
+	if asssumeRoleArn != "" {
+		staticsess, err := session.NewSession(&aws.Config{Credentials: awsConf.Credentials})
+		if err != nil {
+			return nil, err
+		}
+
+		awsConf.Credentials = credentials.NewCredentials(&stscreds.AssumeRoleProvider{
+			Client:   sts.New(staticsess),
+			RoleARN:  asssumeRoleArn,
+			Duration: 15 * time.Minute,
```

Comment:
why 15 minutes? is it based in something?

Reply:
I was using the same value as used here - https://github.com/spiffe/spire/blob/main/pkg/server/plugin/upstreamauthority/awspca/pca_client.go#L39.

As per AWS AssumeRole documentation (https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html), **The value specified can can range from 900 seconds (15 minutes) up to the maximum session duration that is set for the role. The maximum session duration setting can have a value from 1 hour to 12 hours.** 
Since this duration is not configurable, in order to make this work across all scenarios, using the lowest value does seems ok to me. 
As per go library documentation, the credentials are cached and refreshed automatically on expiration.